packages/analytics-types/src/logger.ts

```diff
@@ -4,11 +4,18 @@ export interface Logger {
   log(...args: any[]): void;
   warn(...args: any[]): void;
   error(...args: any[]): void;
+  debug(...args: any[]): void;
 }
 
 export enum LogLevel {
   None = 0,
   Error = 1,
   Warn = 2,
   Verbose = 3,
+  Debug = 4,
```

Comment:
thinking verbose and debug can be one. what do you think? browser and node seems to have varying opinions here

Reply:
I agree to merge debug and verbose. 
<img width="247" alt="image" src="https://user-images.githubusercontent.com/5042222/197876922-b13f657b-1bf1-4c85-a1d5-1d4dcfb52e22.png">

Just curious if the logLevel is set to None, can we still get logger information? It does not need to print on the console, but store somewhere or some extensions can retrieve the logger information.

Reply:
Good question, I added the debug level to match the `debug` level: "for debugging information. These messages are displayed only when user had opted-in to see debug logging for specific modules", https://webpack.js.org/api/logging/.

My plan is to have the following items in interface, with the actually logger implementation to decide.
- error
- warn
- log
- debug

Browser and Node has the `console.debug`, that maps the output to `Verbose` in browser, and `console.log` in Node.

For decision to whether merging debug into log, I am ok with both. Just some concern that `debug` outputs much more information than `log`, might be overwhelming information if user enables `log`.

@dnshi , for your question, no, the logged messages will not be stored/sent to anywhere, see https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-core/src/logger.ts#L34.

Reply:
@liuyang1520 I see. I am wondering if we have a new version of Chrome extension, how the extension gets the message if the logger level is set to None by the customer?

Reply:
Chatted with Marvin offline. The Chrome extension support will be in a separate topic.